### PR TITLE
Add CLI to add/remove permissions from existed role

### LINKED ISSUE #26155: Add CLI to add/remove permissions from existed role
### Body

Followup on https://github.com/apache/airflow/pull/25854
[Roles CLI](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#roles) currently support create, delete, export, import, list

It can be useful to have the ability to add/remove permissions from existed role.
This has also been asked in https://github.com/apache/airflow/issues/15318#issuecomment-872496184

cc @chenglongyan

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by chenglongyan**
closes: #26155

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/cli/commands/role_command.py [hunk 1]
```diff
@@ -17,23 +17,40 @@
...
+        AirflowConsole().print_as(
+            data=sorted(r.name for r in roles), output=args.output, mapper=lambda x: {"name": x}
+        )
+        return
+
+    permission_map = {}
+    for role in roles:
+        for permission in role.permissions:
+            permission_map.setdefault((role.name, permission.resource.name), [])
+            permission_map.get((role.name, permission.resource.name)).append(permission.action.name)
```
Comment (uranusjr) [inline airflow/cli/commands/role_command.py]: Would be easier with defaultdict:

```python
permission_map: Dict[Tuple[str, str], List[str]] = collections.defaultdict(list)
for role in roles:
    for perm in role.permissions:
        permission_map[(role.name, perm.resource.name)].append(perm.action.name)
```

File: airflow/cli/commands/role_command.py [hunk 2]
```diff
@@ -17,23 +17,40 @@
...
+
     AirflowConsole().print_as(
-        data=sorted(r.name for r in roles), output=args.output, mapper=lambda x: {"name": x}
+        data=sorted(p for p in permission_map.keys()),
```
Comment (uranusjr) [inline airflow/cli/commands/role_command.py]:
```suggestion
        data=sorted(permission_map),
```

File: airflow/cli/commands/role_command.py [hunk 3]
```diff
@@ -64,6 +81,69 @@ def roles_delete(args):
...
+            print(f"Role named '{name}' does not exist")
+            exit(1)
+
+        role_map[name] = role
+        for permission in role.permissions:
+            perm_map.setdefault((name, permission.resource.name), set())
+            perm_map[(name, permission.resource.name)].add(permission.action.name)
+
+    for name in args.resource:
+        resource: Resource = asm.get_resource(name)
```
Comment (uranusjr) [inline airflow/cli/commands/role_command.py]: Should this be `Optional[Resource]`? (Same for other calls in this function)

File: airflow/cli/cli_parser.py [hunk 4]
```diff
@@ -2087,7 +2101,10 @@ def get_long_option(arg: Arg):
         """Get long option from Arg.flags"""
         return arg.flags[0] if len(arg.flags) == 1 else arg.flags[1]
 
-    positional, optional = partition(lambda x: x.flags[0].startswith("-"), args)
+    def __flags(x):
+        return x.flags[0].startswith("-")
+
+    positional, optional = partition(__flags, args)
```
Comment (uranusjr) [inline airflow/cli/cli_parser.py]: Why this change? Seems equivalent to me

**Review by uranusjr — APPROVED** [review]

**Comment by eladkal** [comment]
Static checks fails :(

File: airflow/cli/cli_parser.py (hunk 4 above)
Comment (chenglongyan) [reply airflow/cli/cli_parser.py]: emmm，It was modified during debugging and has been adjusted back...

**Review by potiuk — APPROVED** [review]

**Comment by jyotsa09** [comment]
Hi @chenglongyan I was just trying to use this feature but have some confusion. It would be great if you can help me out to understand this. 
While adding/deleting permission for the role we need `--action` and `--resource` as a required arguments but CLI says its an optional arguments. Below is the screenshot-
Screenshot 2022-11-09 at 9 18 10 PM

Also can we add multiple actions/resources at a time ? like this - 
`airflow roles add-perms test-role -a '[can_read,can_edit]' -r 'XComs'`
`airflow roles add-perms test-role -a 'can_read' -r '[XComs, Password]'`

**Comment by chenglongyan** [comment]
@jyotsa09 you can do like this.

> `airflow roles add-perms test-role -a can_read can_edit -r XComs`
   `airflow roles add-perms test-role -a can_read -r XComs Password`
